### PR TITLE
[pkg/ottl] add string_converter

### PR DESCRIPTION
**Pull request by led0nk**
**Description:** 
- implemented func_string (string_converter)
- added new unit tests in `pkg/ottl/ottlfuncs/func_string_test.go`
- added function to `pkg/ottl/ottlfuncs/functions.go`
- updatet README.md at `pkg/ottl/ottlfuncs/README.md`
- added changelog

**Link to tracking Issue:** 
- #27867 

**Testing:** 

- added similar tests as in `pkg/ottl/ottlfuncs/func_double_test.go`

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .chloggen/string-converter.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: "enhancement"
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: stringconverter
```
Comment (TylerHelmuth) [inline .chloggen/string-converter.yaml:7]:
```suggestion
component: pkg/ottl
```

File: pkg/ottl/ottlfuncs/README.md
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:1]: Please remove the extra formatting and only add the new function doc

Comment (led0nk) [reply pkg/ottl/ottlfuncs/README.md:1]: reversed extra formatting by autoformat in new commit, sorry for that

File: .chloggen/string-converter.yaml (hunk 1 above)
Comment (led0nk) [reply .chloggen/string-converter.yaml]: done in new commit

File: pkg/ottl/ottlfuncs/README.md [hunk 2]
```diff
@@ -1068,6 +1069,32 @@ Examples:
...
+
+The `String` Converter converts the `value` to string type.
+
+The returned type is `string`.
+
+- float64. The function returns the `value` as a string type.
+- string. The function returns the `value`, even if it's empty.
+- bool. The function returns true/false as "true"/"false".
+- int64. The function returns the `value` as a string type.
+- struct. The function returns an empty struct as "{}"
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:1084]: This isn't true.  Checkout https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/16378865457151125248414e887edf62a507abae/pkg/ottl/expression.go#L389-L422 for how the conversion happens

**Review by TylerHelmuth — COMMENTED** [review]
Please add a test case to `pkg/ottl/e2e`

File: pkg/ottl/ottlfuncs/README.md (hunk 2 above)
Comment (led0nk) [reply pkg/ottl/ottlfuncs/README.md]: i updated that in latest commit

**Comment by led0nk** [comment]
updated README - added e2e test cases for string-conversion and updated the function test :)

**Review by TylerHelmuth — APPROVED** [review]

**Comment by led0nk** [comment]
accidentally deleted the branch, sorry for that :sweat_smile:

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**